codec-http2/src/main/java/io/netty/handler/codec/http2/DelegatingDecompressorFrameListener.java

```diff
@@ -402,52 +368,56 @@ EmbeddedChannel decompressor() {
         }
 
         /**
-         * Increment the number of decompressed bytes processed by the application.
-         */
-        void incrementProcessedBytes(int delta) {
-            if (processed + delta < 0) {
-                throw new IllegalArgumentException("processed bytes cannot be negative");
-            }
-            processed += delta;
-        }
-
-        /**
          * Increment the number of bytes received prior to doing any decompression.
          */
         void incrementCompressedBytes(int delta) {
-            if (compressed + delta < 0) {
-                throw new IllegalArgumentException("compressed bytes cannot be negative");
-            }
+            assert delta >= 0;
             compressed += delta;
+            if (compressed < 0) {
+                throw new IllegalStateException("compressed bytes cannot be negative. compressed: " +
+                        compressed + " delta: " + delta);
+            }
         }
 
         /**
-         * Increment the number of bytes after the decompression process. Under normal circumstances this
-         * delta should not exceed {@link Http2Decompressor#processed)}.
+         * Increment the number of bytes after the decompression process.
          */
         void incrementDecompressedByes(int delta) {
-            if (decompressed + delta < 0) {
-                throw new IllegalArgumentException("decompressed bytes cannot be negative");
-            }
+            assert delta >= 0;
             decompressed += delta;
+            if (decompressed < 0) {
```

Comment:
I guess this is no longer needed now that it is being used strictly to increment ... overflow should be guarded by flow control.